tests/test_lib_tokens_push.py

```diff
@@ -392,6 +393,44 @@ def test_03_api_authenticate_client(self):
             # Result-Value is True, since the challenge is marked resolved in the DB
         self.assertTrue(jsonresp.get("result").get("value"))
 
+        # We mock the ServiceAccountCredentials, since we can not directly contact the Google API
+        # Do single shot auth with waiting
+        with mock.patch('privacyidea.lib.smsprovider.FirebaseProvider.ServiceAccountCredentials') as mySA:
+            # alternative: side_effect instead of return_value
+            mySA.from_json_keyfile_name.return_value = myCredentials(myAccessTokenInfo("my_bearer_token"))
+
+            # add responses, to simulate the communication to firebase
+            responses.add(responses.POST, 'https://fcm.googleapis.com/v1/projects/test-123456/messages:send',
+                          body="""{}""",
+                          content_type="application/json")
+
+            # In two seconds we need to run an update on the challenge table.
+            Timer(2, self.mark_challenge_as_accepted).start()
+
+            set_policy("push1", scope=SCOPE.AUTH, action="{0!s}=20".format(PUSH_ACTION.WAIT))
+            # Send the first authentication request to trigger the challenge
+            with self.app.test_request_context('/validate/check',
+                                               method='POST',
+                                               data={"user": "cornelius",
+                                                     "realm": self.realm1,
+                                                     "pass": "pushpin"}):
+                res = self.app.full_dispatch_request()
+                self.assertTrue(res.status_code == 200, res)
+                jsonresp = json.loads(res.data.decode('utf8'))
+                # We successfully authenticated! YEAH!
+                self.assertTrue(jsonresp.get("result").get("value"))
+                self.assertTrue(jsonresp.get("result").get("status"))
+                self.assertEqual(jsonresp.get("detail").get("serial"), tokenobj.token.serial)
```

Comment:
What do you think about adding a test for the case that the challenge is not marked as accepted during the timeout? Maybe with a shorter timeout then, like 1 or 2 seconds :)

... and maybe also for the case that the PIN is not correct.

Reply:
Will do

Reply:
done in d370c8f9